test/cmp_hdr_test.c

```diff
@@ -343,13 +346,11 @@ execute_HDR_generalInfo_push1_items_test(CMP_HDR_TEST_FIXTURE *fixture)
     if (!TEST_int_eq(ASN1_TYPE_cmp(itav->infoValue.other, pushed_val), 0))
         goto err;
 
-    sk_OSSL_CMP_ITAV_pop_free(itavs, OSSL_CMP_ITAV_free);
-    return 1;
+    res = 1;
+
  err:
-    ASN1_INTEGER_free(asn1int);
-    ASN1_TYPE_free(val);
```

Comment:
I suppose there is one more issue with this test - if OSSL_CMP_ITAV_create() fails the val will leak. Not sure if OSSL_CMP_ITAV_push0_stack_item can fail - its return value is not tested.

Reply:
It can fail.  I'll add a check.

Reply:
You did not address the val leak if OSSL_CMP_ITAV_create() fails

Reply:
It's late :(